heatpump.yaml

```diff
@@ -2821,12 +2702,500 @@ switch:
+            id(unmasked_value_register_211) = new_value;
+            esphome::modbus_controller::ModbusCommandItem set_payload_command = esphome::modbus_controller::ModbusCommandItem::create_write_single_command(${devicename}, 211, new_value);
+            ${devicename}->queue_command(set_payload_command);
+          }
+    on_turn_off:
+      #- logger.log: "Switch Turned Off!"    
+      - lambda: |-
+          uint16_t checked_bit = 0x40;
+          uint16_t new_value = id(unmasked_value_register_211);  // The original unmasked value
+          new_value &= ~checked_bit;                             // Clear and set
+          if ((new_value) != id(unmasked_value_register_211)) {
+            id(unmasked_value_register_211) = new_value;
+            esphome::modbus_controller::ModbusCommandItem set_payload_command = esphome::modbus_controller::ModbusCommandItem::create_write_single_command(${devicename}, 211, new_value);
+            ${devicename}->queue_command(set_payload_command);
+          }
+  # Register: 211, Bit: 7, default: false, TODO: verify default
+  - platform: template
+    name: "Parameter Setting 2 Smart Grid"
+    id: "${devicename}_parameter_setting_2_smart_grid"
+    icon: mdi:eye
+    #bitmask: 0x80
+    restore_mode: DISABLED
+    entity_category: config
+    optimistic: true
+    lambda: "return (id(unmasked_value_register_211) & 0x80) == 0x80;"
+    on_turn_on:
+      - lambda: |-
+          uint16_t checked_bit = 0x80;
+          uint16_t new_value = id(unmasked_value_register_211);  // The original unmasked value
+          new_value &= ~checked_bit;                             // Clear and set
+          new_value += checked_bit;
+          if ((new_value) != id(unmasked_value_register_211)) {
+            id(unmasked_value_register_211) = new_value;
+            esphome::modbus_controller::ModbusCommandItem set_payload_command = esphome::modbus_controller::ModbusCommandItem::create_write_single_command(${devicename}, 211, new_value);
+            ${devicename}->queue_command(set_payload_command);
+          }
+    on_turn_off:
+      - lambda: |-
+          uint16_t checked_bit = 0x80;
+          uint16_t new_value = id(unmasked_value_register_211);  // The original unmasked value
+          new_value &= ~checked_bit;                             // Clear and set
+          if ((new_value) != id(unmasked_value_register_211)) {
+            id(unmasked_value_register_211) = new_value;
+            esphome::modbus_controller::ModbusCommandItem set_payload_command = esphome::modbus_controller::ModbusCommandItem::create_write_single_command(${devicename}, 211, new_value);
+            ${devicename}->queue_command(set_payload_command);
+          }
+  # Register: 211, Bit: 8, default: false, TODO: verify default
+  - platform: template
+    name: "Parameter Setting 2 Port Definition"
+    id: "${devicename}_parameter_setting_2_port_definition"
+    disabled_by_default: true
```

Comment:
Let's not disable this one. It is up to the user if he/she wants to set in a dashboard, but imho we should enable these so that (historic) metrics are always available.

Reply:
ok
this don’t disable completely. it makes not default visible values in home assistant dashboard, but in home assistant in esp home sensor edit it is possible to enable it from that time is normally recorded sensor in home assistant;)